### DISCUSSION TITLE
GUI not accepting uppercase characters during object creation

### DISCUSSION THREAD

**Discussion by arkhnchul** [dbeaver/dbeaver#19110] [Q&A]
### Description

If the object (table, column, etc) name contains uppercase letters during creation, it converts to lowercase after "OK" button is pressed. Editing already existing (e.g. persisted in the database) objects works as expected.

### DBeaver Version

community 22.3.4

### Operating System

Debian linux unstable

### Steps to reproduce

1. Try to create any object (table, column)
2. Type uppercase letter in the "Name" field, press OK
3. It will be converted to lowercase

![2023-02-17_21_18_34](https://user-images.githubusercontent.com/17163127/219751818-058d39f6-24e3-4d44-a520-594332959494.jpg)
![2023-02-17_21_18_43](https://user-images.githubusercontent.com/17163127/219751829-53bdb543-45f8-49e5-be60-209b7964be01.jpg)
![2023-02-17_21_20_16](https://user-images.githubusercontent.com/17163127/219751834-d9d64d0e-817b-455d-ac25-c5159382e750.jpg)

**Comment by E1izabeth** [comment]
If I correctly understand you, it's not a bug. PostgreSQL fold everything to lower case. It's an intended behaviour of this databease. I'm sure, that if you save an object after editing and update your metadata, it will be in lower case unless you add quotes to the identifier name.

**Comment by arkhnchul** [comment]
if the object is already exists, it adds quotes automagically, just as  expected
![2023-02-17_21_42_53](https://user-images.githubusercontent.com/17163127/219757947-dcfc5664-b64d-464d-bebd-62c2c1c1602d.jpg)

**Comment by E1izabeth** [comment]
Thank you for quick reply. It became clearer now.

**Comment by LonwoLonwo** [comment]
Hello @arkhnchul 

This mechanism works as expected and does not quote names without special setting.

![2023-02-18 21_02_59-Preferences](https://user-images.githubusercontent.com/45152336/219881412-d165462e-a0c2-4b93-a08f-078067a7d4d5.png)

But you can change this behaviour by enabling in Preferences -> Connections -> Metadata the `"Use case-sensitive names" ` setting

**Comment by arkhnchul** [comment]
thanks, did not know that there is a setting for this. The behaviour is not buggy then, just inconsistent - when renaming existing objects case-sensitive names are accepted with correct quoting with that setting turned off, which was the source of sonfusion

**Comment by LonwoLonwo** [comment]
DBeaver supports many different databases with different cases. 
We do not quote object names for object creation because databases usually transform these names into database cases from their side. But yes, we do it only for this case because this is the most popular operation (object creation).
And I do not think this is inconsistent - ignore quotes in the renaming case.
You can add quotes in the object name yourself as a second workaround.
